lib/workers/global/config/parse/host-rules-from-env.ts

```diff
@@ -23,9 +49,16 @@ export function hostRulesFromEnv(env: NodeJS.ProcessEnv): HostRule[] {
     // Double underscore __ is used in place of hyphen -
     const splitEnv = envName.toLowerCase().replace(/__/g, '-').split('_');
     const hostType = splitEnv.shift()!;
-    if (datasources.has(hostType)) {
-      const suffix = splitEnv.pop()!;
-      if (isAuthField(suffix)) {
+    if (
+      datasources.has(hostType) ||
+      (platforms.has(hostType) && splitEnv.length > 1)
+    ) {
+      let suffix = splitEnv.pop()!;
+      if (isAuthField(suffix) || isHttpsAuthField(suffix)) {
+        if (isHttpsAuthField(suffix)) {
+          suffix = restoreHttpsAuthField(suffix);
+        }
+
         let matchHost: string | undefined = undefined;
         const rule: HostRule = {};
         rule[suffix] = env[envName];
```

Comment:
we should only copy if `env[envName]` isn't undefined.

Reply:
that's done in the `setHostRuleValue` function